segfast/utils.py

```diff
@@ -35,6 +37,74 @@ def update(self, n=1):
 notifier = Notifier
 
 
+class TraceHeaderSpec:
+    """ Trace header class to store its name and byte position. By default, byte position is defined by name
+    accordingly to SEG-Y specification.
+
+    Parameters
+    ----------
+    name : str
+        Name of the header.
+    start_byte : int, optional
+        Byte position of the header, by default None. If None, default byte position from the spec will be used.
+    dtype : int, str or dtype, optional
+        dtype for header (e.g. 'i2', '>f4', `np.float32`) or its length in bytes (then is interpreted as integer type).
+    byteorder : '>' or '<', optional
+        Endianness to use, if it's not defined by dtype. If None and dtype doesn't specify, architecture default
+        will be used.
+    """
+    TRACE_HEADER_SIZE = 240
+
+    STANDARD_HEADER_TO_BYTE = segyio.tracefield.keys
+    STANDARD_BYTE_TO_HEADER = {v: k for k, v in STANDARD_HEADER_TO_BYTE.items()}
+
+    START_BYTES = sorted(STANDARD_HEADER_TO_BYTE.values())
+    STANDARD_BYTE_TO_LEN = {start: end - start
+                            for start, end in zip(START_BYTES, START_BYTES[1:] + [TRACE_HEADER_SIZE + 1])}
+
+    def __init__(self, name=None, start_byte=None, dtype=None, byteorder=None):
+        self.name = name or self.STANDARD_BYTE_TO_HEADER[start_byte]
+        self.start_byte = start_byte or self.STANDARD_HEADER_TO_BYTE[name]
+
+        dtype = dtype or self.STANDARD_BYTE_TO_LEN[self.start_byte]
+        if isinstance(dtype, int):
+            dtype = 'i' + str(dtype)
+
+        self.dtype = np.dtype(dtype)
+        if isinstance(dtype, str) and dtype[0] not in {'>', '<'} and byteorder is not None:
```

Comment:
Which is even more correct: not `np.int32` will have a properly set `byteorder` if given
```suggestion
        self.has_explicit_byteorder = isinstance(dtype, str) and dtype[0] in {'>', '<'}
        if not self.has_explicit_byteorder and byteorder is not None:
```